cmd/bosun/expr/expr.go

```diff
@@ -212,6 +212,20 @@ func (r ResultSlice) DescByValue() ResultSlice {
 	return c
 }
 
+// Filter strips out results if their tagset doesn't conform to the given key/value pair restrictions
+// It updates the input slice in place.
+func (r ResultSlice) Filter(filter opentsdb.TagSet) ResultSlice {
+	lastGoodIndex := -1
+
+	for _, res := range r {
+		if res.Group.Compatible(filter) {
+			lastGoodIndex++
+			r[lastGoodIndex] = res
```

Comment:
How about we just make Filter return a new ResultSlice instead of the in place stuff?